regression/synthetic_dataset.py

```diff
@@ -260,9 +267,9 @@ def main(args: Namespace, kwargs: Dict[str, str]) -> None:
     """
     dataset_params = get_dataset_params(kwargs)
     writer_params = get_writer_params(kwargs)
-    if args.name.lower() not in DATASET_MAP:
-        raise ValueError(f'Unsupported dataset {args.name}. Supported: {DATASET_MAP.keys()}')
-    dataset = DATASET_MAP[args.name.lower()](**dataset_params)
+    if args.name.lower() not in _DATASET_MAP:
+        raise ValueError(f'Unsupported dataset {args.name}. Supported: {_DATASET_MAP.keys()}')
+    dataset = getattr(sys.modules[__name__], _DATASET_MAP[args.name.lower()])(**dataset_params)
```

Comment:
Why do you need this getattr + sys.modules? Does `dataset = _DATASET_MAP[args.name.lower()](**dataset_params)` not work?

Reply:
That's because the values are a string and not a class name. If you put values as a class name then it will say class is not defined since those are getting initialized at later point.